### PR TITLE
Test improved input validation errors

### PR DESCRIPTION
**Pull request by indrajit96**
Added new test cases and models for better input validation.
DLIS-6165

### CODE REVIEW

**Comment by rmccorm4** [comment]
Tagging @kthui @lkomali to help review

**Comment by rmccorm4** [comment]
@indrajit96 make sure to address the codeql checks above. Let's also rename the test folder to `L0_input_validation` to match `L0_output_validation` and not confuse "val" with "value"

**Comment by indrajit96** [comment]
> @indrajit96 make sure to address the codeql checks above. Let's also rename the test folder to `L0_input_validation` to match `L0_output_validation` and not confuse "val" with "value"

Done

File: qa/L0_input_validation/input_validation_test.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                model_name="input_all_required",
+                inputs=inputs,
+            )
+        err_str = str(e.exception)
+        self.assertIn(
+            "expected 3 inputs but got 1 inputs for model 'input_all_required'. Got input(s) ['INPUT0'], but missing required input(s) ['INPUT1','INPUT2']. Please provide all required input(s).",
+            err_str,
+        )
+
+    def test_input_validation_optional(self):
```
Comment (rmccorm4) [inline qa/L0_input_validation/input_validation_test.py:87]: I think adding another test for a model with **only** optional inputs would be good to make sure we're not accidentally raising an error on models that don't require any inputs.

File: qa/L0_input_validation/input_validation_test.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+# Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (rmccorm4) [inline qa/L0_input_validation/input_validation_test.py:2]:
```suggestion
# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```

File: qa/L0_input_validation/models/input_all_required/1/model.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Copyright 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (rmccorm4) [inline qa/L0_input_validation/models/input_all_required/1/model.py:1]: Please check the copyrights of all files
```suggestion
# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```

File: qa/L0_input_validation/models/input_all_required/config.pbtxt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+output [
+  {
+    name: "OUTPUT0"
+    data_type: TYPE_FP32
+    dims: [ 1 ]
+  }
+]
+
+instance_group [{ kind: KIND_CPU }]
```
Comment (kthui) [inline qa/L0_input_validation/models/input_all_required/config.pbtxt:55]: Same, leave an empty line at the end of a file.

File: qa/L0_input_validation/models/input_optional/config.pbtxt [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+output [
+  {
+    name: "OUTPUT0"
+    data_type: TYPE_FP32
+    dims: [ 1 ]
+  }
+]
+
+instance_group [{ kind: KIND_CPU }]
```
Comment (kthui) [inline qa/L0_input_validation/models/input_optional/config.pbtxt:61]: Same, leave an empty line at the end of a file.

File: qa/L0_input_validation/models/input_all_optional/config.pbtxt [hunk 6]
```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
```
Comment (kthui) [inline qa/L0_input_validation/models/input_all_optional/config.pbtxt:1]: Please always use "NVIDIA CORPORATION & AFFILIATES" for any new files or when updating years on existing files, i.e. "Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved."

We used to just say "NVIDIA CORPORATION", but there was an update years back that switched over to "NVIDIA CORPORATION & AFFILIATES".

File: qa/L0_input_validation/models/input_all_required/config.pbtxt [hunk 7]
```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
```
Comment (kthui) [inline qa/L0_input_validation/models/input_all_required/config.pbtxt:1]: Same, "NVIDIA CORPORATION & AFFILIATES".

File: qa/L0_input_validation/models/input_optional/config.pbtxt [hunk 8]
```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
```
Comment (kthui) [inline qa/L0_input_validation/models/input_optional/config.pbtxt:1]: Same, "NVIDIA CORPORATION & AFFILIATES"

**Review by kthui — APPROVED** [review]
LGTM!

**Review by rmccorm4 — APPROVED** [review]
Nice work!

**Pull request merged**